jetson/nav/search/searchStateMachine.cpp

```diff
@@ -259,18 +305,56 @@ NavState SearchStateMachine::executeDriveToTarget()
     double distance = mRover->roverStatus().leftTarget().distance;
     double bearing = mRover->roverStatus().leftTarget().bearing + mRover->roverStatus().odometry().bearing_deg;
 
+    // Update if we have an empty leftTarget, BUT we have a valid cached leftTarget
+    if ( mRover->roverStatus().leftTarget().distance < 0 && mRover->roverStatus().leftCacheTarget().distance != -1 ) {
+        distance = mRover->roverStatus().leftCacheTarget().distance;
+        bearing = mRover->roverStatus().leftCacheTarget().bearing + mRover->roverStatus().odometry().bearing_deg;
+        cout << "USING CACHE" << endl;
+    }
+
     // Executes the logic for driving with 0, 1, or 2 targets in sight
     // If we have a second target detected, determine which post is closer
     // If the distance to the second target is less than the first,
     // set our variables to the target 2's distance and bearing
     // Else, use the initialized values from target 1 when driving
     if( mRover->roverStatus().rightTarget().distance > 0 )
```

Comment:
Shouldn't this if and else if statement be inside of the if statement on line 309, if you will only have a valid right tag if there is a valid left tag?